src/mbed_tools/project/_internal/git_utils.py

```diff
@@ -41,18 +41,18 @@ def clone(url: str, dst_dir: Path) -> git.Repo:
         raise VersionControlError(f"Cloning git repository from url '{url}' failed. Error from VCS: {err.stderr}")
 
 
-def checkout(repo: git.Repo, ref: str, force: bool = False) -> None:
+def deploy(repo: git.Repo, ref: str, force: bool = False) -> None:
     """Check out a specific reference in the given repository.
 
     Args:
-        repo: git.Repo object where the checkout will be performed.
+        repo: git.Repo object where the deploy will be performed.
         ref: Git commit hash, branch or tag reference, must be a valid ref defined in the repo.
 
     Raises:
         VersionControlError: Check out failed.
     """
     try:
-        repo.git.checkout(f"--force {ref}" if force else str(ref))
+        repo.git.deploy(f"--force {ref}" if force else str(ref))
```

Comment:
This isn't going to work. `repo.git.checkout` is a call to the `gitpython` library. The parent function doesn't make sense as `deploy` either, this is a `git_utils` module and `git` doesn't have a `deploy` command. We should only need to change the CLI command handler and external name as that's the only place `deploy` makes sense.